### PR TITLE
Refactor criterias class and add a new class CriteriaSearchVideo

### PR DESCRIPTION
**Pull request by melvicaballero**
Refactor criteria family and add a new CriteriaSearchVideo class.

### CODE REVIEW

File: src/main/java/com/jala/common/models/Criteria.java [hunk 1]
```diff
@@ -42,6 +42,11 @@ public Criteria(String pathOrigin) {
         this.pathOrigin = pathOrigin;
     }
 
+    /**
+     * Constructor Criteria class, without arguments.
+     */
+    public Criteria(){};
```
Comment (AKatrin) [inline src/main/java/com/jala/common/models/Criteria.java]: Add space between ")" and "{"

File: src/main/java/com/jala/search/models/CriteriaSearch.java [hunk 2]
```diff
@@ -78,14 +80,20 @@
...
      */
     private int type;
 
     /**
-     * Constructor of class CriteriaSearch.
+     * Constructor of the CriteriaSearch class.
+     */
+    public CriteriaSearch() {
+        LoadCriteria();
+    }
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/CriteriaSearch.java:93]: Add a empty line after "{"

File: src/main/java/com/jala/search/models/CriteriaSearchVideo.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+package com.jala.search.models;
+
+/**
+ * CriteriaSearchVideo class.
+ * @author Melvi Caballero.
+ * @version 0.0.1
+ */
+public class CriteriaSearchVideo extends CriteriaSearch {
+    private String path;
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/CriteriaSearchVideo.java]: Add documentation to the attributes

File: src/main/java/com/jala/search/models/CriteriaSearchVideo.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    private String aspecRatio;
+    private String dimension;
+
+    /**
+     * Constructor of CriteriaSearchVideo class.
+     */
+    public CriteriaSearchVideo() {
+        super();
+        LoadVideoCriteria();
+
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/CriteriaSearchVideo.java]: Delete empty line

File: src/main/java/com/jala/search/models/CriteriaSearchVideo.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    /**
+     * Load video criteria goes to init the empty values.
+     */
+    private void LoadVideoCriteria(){
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/CriteriaSearchVideo.java]: The name methods should start with lowercase

File: src/main/java/com/jala/search/models/CriteriaSearchVideo.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        this.setAudioCodec("");
+        this.setFrameRate("");
+        this.setAspecRatio("");
+        this.setDimension("");
+    }
+    /**
+     * Constructor CriteriaSearchVideo class with a path parameter.
+     * @param path the path of file.
+     */
+    public CriteriaSearchVideo(String path) {
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/CriteriaSearchVideo.java]: The name methods should start with lowercase

File: src/main/java/com/jala/search/models/CriteriaSearchVideo.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        this.LoadVideoCriteria();
+        this.path = path;
+    }
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/CriteriaSearchVideo.java]: Add empty line after "}"

File: src/main/java/com/jala/search/models/CriteriaSearchVideo.java [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+    /**
+     * @return dimension get dimension attribute of video search.
+     */
+    public String getDimension() {
+        return dimension;
+    }
+
+    /**
+     * @param dimension set dimension attribute os video search.
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/CriteriaSearchVideo.java]: change "os" to "of"

File: src/main/java/com/jala/search/models/CriteriaSearchVideo.java [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+ * CriteriaSearchVideo class.
+ * @author Melvi Caballero.
+ * @version 0.0.1
+ */
+public class CriteriaSearchVideo extends CriteriaSearch {
+    private String path;
+    private String videoCodec;
+    private String audioCodec;
+    private String frameRate;
+    private String aspecRatio;
```
Comment (AKatrin) [inline src/main/java/com/jala/search/models/CriteriaSearchVideo.java]: Rename "aspecRatio" to "aspectRatio"

**Review by AKatrin — CHANGES_REQUESTED** [review]

File: src/main/java/com/jala/search/models/CriteriaSearchVideo.java (hunk 6 above)
Comment (melvicaballero) [reply src/main/java/com/jala/search/models/CriteriaSearchVideo.java]: This method is a constructor.

File: src/main/java/com/jala/search/models/CriteriaSearchMultimedia.java [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+     */
+    public CriteriaSearchMultimedia(String path) {
+        super(path);
+        this.LoadVideoCriteria();
+    }
+
+    /**
+     * Load video criteria goes to init the empty values.
+     */
+    private void LoadVideoCriteria() {
```
Comment (Regis18) [inline src/main/java/com/jala/search/models/CriteriaSearchMultimedia.java]: The first letter of the method has to be lower case... loadVideoCriteria

**Review by Regis18 — APPROVED** [review]

**Pull request merged**